src/Fixture.hpp

```diff
@@ -132,6 +151,76 @@ namespace iodrivers_base
         {
             return driver.getStatus().queued_bytes;
         }
+        void EXPECT_REPLY(std::vector<uint8_t> const& expectation, std::vector<uint8_t> const& reply)
+        {
+            getStream()->EXPECT_REPLY(expectation,reply);
+        }
+        
+        /**
+         * Check if the test has any expectation set and throw if positive.
+         * It should be used to check if the test reached its end without 
+         * any expecation left only
+         */
+        void expectationsIsEmpty()
+        {
+            if(!getStream()->expectationsIsEmpty())
+                throw TestEndsWithExcepetionsLeftException();
+        }
+        
+        void enableMockMode()
+        {
+            getStream()->enableMockMode();
+        }
+        
+        void clearExpectations()
+        {
+            getStream()->clearExpectations();
+        }
+        
+        /** 
+          * GTEST FAIL assertion can only be used in void-returning functions.
+          * Constructors and Destructors are not considered void-returning functions, 
+          * according to the C++ language specification, and so you may not use fatal assertions in them.
+          * Using a Fatal assertion on these method would leave the object in a partially state. 
+          * I was then decided to use the tear down to finalize the tests in GTEST.
+          * In BOOST the teardown method can be normally called in the destructor method
+         */
+        void tearDownMock()
+        {
+            try
+            {
+                expectationsIsEmpty();
+            }
+            catch(TestEndsWithExcepetionsLeftException e)    
+            {
+                #ifdef IODRIVERS_BASE_FIXTURE_GTEST_FRAMEWORK
+                ADD_FAILURE() << "IODRIVERS_BASE_MOCK Error: Test reached its end without satisfying all expecations.";
+                #endif
+                #ifdef IODRIVERS_BASE_FIXTURE_BOOST_FRAMEWORK
+                BOOST_ERROR("IODRIVERS_BASE_MOCK Error: Test reached its end without satisfying all expecations.");
+                #endif
+            }
+        }
+        
+        class MockContext
+        { 
+        public:
+            MockContext() {};
```

Comment:
Why the default constructor ?

This is plain wrong, actually. You end up with a pointer that is not initialized. Yuk.